### PR TITLE
merging  tonyear code into new tonyear repository

### PR DESCRIPTION
**Pull request by freyac**
Updating the project template per Joe's checklist!

### CODE REVIEW

File: carbonplan_tonyear/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import pathlib
+
+from intake import open_catalog
+
+cat_dir = pathlib.Path(__file__)
+cat_file = str(cat_dir.parent / "data/catalog.yaml")
+cat = open_catalog(cat_file)
```
Comment (jhamman) [inline carbonplan_tonyear/__init__.py:7]:
```suggestion
from .core import calculate_tonyears
```

File: README.md [hunk 2]
```diff
@@ -3,29 +3,16 @@
...
-
-`scripts` and `tests` contain filler .py files. Update/remove these with your project name.
-### Updating workflows/main.yaml
-
-In the workflows/main.yaml file, the pytest and docker sections of the github actions configuration are currently commented out. If you wish to add them, uncomment them.
-
-### Updating requirements.txt
-
-requirements.txt is currently empty. You can populate it with: ```pip3 freeze > requirements.txt```
+This repository is a sandbox for exploring and understanding ton-year accounting and related issues. We're working on replicating three ton-year accounting methods in `notebooks/tonyear.ipynb`, which is likely a good place to start poking around.
```
Comment (jhamman) [inline README.md:15]: I can't comment on line 23 below but there's one more reference to the template that needs to be updated

File: carbonplan_tonyear/tests/test_catalog.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import os
+
+from carbonplan_project import cat_file
+
+
+def test_catalog_exists():
+    assert os.path.exists(cat_file)
```
Comment (jhamman) [inline carbonplan_tonyear/tests/test_catalog.py:7]:
```suggestion
from carbonplan_tonyear import calculate_tonyears


def test_calculate_tonyears):
    calculate_tonyears(...insert a sample set of arguments here...)
```

**Review by jhamman — COMMENTED** [review]
The last thing to do here is to run pre-commit on the files. We need to add some basic docs on how to set this up but here are the steps:

1. install pre-commit (`pip install pre-commit`)
2. install the pre-commit hooks
    ```
    cd ton-year  # top level of the repo
    pre-commit install
    ```
3. Run pre-commit on all files `pre-commit run --all-files`

This will touch up formatting issues and keep the code style consistent with our other projects.

File: README.md (hunk 2 above)
Comment (freyac) [reply README.md:15]: got it!

File: carbonplan_tonyear/__init__.py (hunk 1 above)
Comment (freyac) [reply carbonplan_tonyear/__init__.py:7]: So, my last hurdle is that the imports in `carbonplan_tonyear/__init__.py` are raising pre-commit flake8 fail. 

> carbonplan_tonyear/__init__.py:1:1: F401 '.core.calculate_tonyears' imported but unused

How to navigate?

Comment (freyac) [reply carbonplan_tonyear/__init__.py:7]: Telling flake8 to ignore imports in `__init__.py` with `# flake8: noqa`.

File: carbonplan_tonyear/core.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    print()
+    print('Discount rate: ' + discount + '%')
+    print('Delay: ' + delay + ' year(s)')
+    print('Baseline atmospheric cost: ' + baseline_atm_cost + ' ton-years')
+    print('Benefit from 1tCO2 with delay: ' + benefit + ' ton-years')
+    print('Number needed: ' + num_needed)
+    print()
+
+
+def calculate_tonyears(method, baseline, time_horizon, delay, discount_rate):
```
Comment (badgley) [inline carbonplan_tonyear/core.py:56]: the comment makes this clear, but just going to ask one more question. 

this function requires that you provide `baseline`, which is the output of `get_irf_curve`. Is it at all useful to sync the terminology here? E.g., `get_baseline_curve`? 

if this is a good idea to change the name, we should update instances of `irf` to `baseline`

File: carbonplan_tonyear/core.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        time_horizon – integer specifying the period over which the impact of an emission
+        is considered (years)
+        delay – integer specifying the emission delay for which a ton-year benefit will be
+        calculated (years)
+        discount_rate – float specifying the discount rate to apply time preference to both
+        costs and benefits over the time horizon
+
+    Function returns a dictionary with:
+        parameters – key parameters used for the calculation
+        baseline – array modeling baseline emission curve (discounted if applicable)
```
Comment (badgley) [inline carbonplan_tonyear/core.py:74]: this is a great doc string. 

i had a question about the output value of `baseline`, wondering if you'd intended for it to be the discounted baseline vs the 'raw' baseline. this answered my question!!!

File: requirements.txt [hunk 6]
```diff
@@ -1,7 +1 @@
-carbonplan
-carbonplan[styles]
-carbonplan[data]
-zarr
-pathlib
-intake
-intake-xarray
+numpy
```
Comment (badgley) [inline requirements.txt:1]: [batteries included](https://docs.python.org/3/library/)

**Review by badgley — COMMENTED** [review]
This looks really great. 

I had one question about naming -- but naming is hard and it's up to you at this point! no bad choices. 

i also removed a few modules requirements.txt that are included in the standard library. Recall  pep8 -- we import standard library, 3rd party modules, our modules -- each separated by whitespace. that means that in requirements we just need to think about those 3rd party libraries!

File: carbonplan_tonyear/core.py (hunk 4 above)
Comment (freyac) [reply carbonplan_tonyear/core.py:56]: I think this a great idea! Changing.

**Pull request merged**